modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -403,29 +403,50 @@ object GmosControllerEpics extends GmosEncoders {
         sys.dhsConnected.map(_.trim === DhsConnected).ifM(IO.unit,
           IO.raiseError(SeqexecFailure.Execution("GMOS is not connected to DHS")))
 
-      override def stopObserve: IO[Unit] =
-        L.info("Stop Gmos exposure") *>
-          sys.stopCmd.setTimeout[IO](DefaultTimeout)
-          sys.stopCmd.mark[IO] *>
-          sys.stopCmd.post[IO]
-
-      override def abortObserve: IO[Unit] =
-        L.info("Abort Gmos exposure") *>
-          sys.abortCmd.setTimeout[IO](DefaultTimeout) *>
-          sys.abortCmd.mark[IO] *>
-          sys.abortCmd.post[IO].void
+      // Remaining time when it is not safe to stop, pause or abort
+      val SafetyCutoff = 3.0 // seconds
+
+      override def stopObserve: IO[Unit] = (sys.dcIsAcquiring, sys.countdown).mapN { case (isAcq, timeLeft) =>
+        if(!isAcq)
```

Comment:
It seems there isa common logic across observe/stop/abort, should it be factored out?

Reply:
Done.